notification/src/postgresRepository.ts

```diff
@@ -0,0 +1,70 @@
+import { Pool, PoolConfig, PoolClient, QueryResult } from 'pg'
+
+import { sleep } from './sleep'
+
+export default class PostgresRepository {
+  private connectionPool?: Pool = undefined
+
+  /**
+   * Initializes the connection to the database.
+   * @param poolConfig: configuration object for the connection pool
+   * @param retries:  Number of retries to connect to the database
+   * @param backoffMs:  Time in ms to backoff before next connection retry
+   * @returns reject promise on failure to connect
+   */
+  public async init (poolConfig: PoolConfig, retries: number, backoffMs: number): Promise<void> {
+    this.connectionPool = await this.connectWithRetry(poolConfig, retries, backoffMs)
+    console.info('Successfully established connection to database.')
+  }
+
+  private async connectWithRetry (poolConfig: PoolConfig, retries: number, backoffMs: number): Promise<Pool> {
+    console.debug(`Connecting to database with config:\n${JSON.stringify(poolConfig)}`)
+
+    let lastError: Error | undefined
+    for (let i = 1; i <= retries; i++) {
+      try {
+        return await this.connect(poolConfig)
+      } catch (error) {
+        lastError = error
+        console.info(`Error connecting to database (${i}/${retries})`)
+      }
+      await sleep(backoffMs)
+    }
+    // can be simplified in one line when https://github.com/typescript-eslint/typescript-eslint/issues/2642 is fixed
+    const error = lastError ?? new Error('Failed to connect to database')
```

Comment:
throw lastError ?? new Error('Failed to connect to database'). 
Jonas fixed it :D

Reply:
Hm, kinda still get the error :(

![image](https://user-images.githubusercontent.com/28054628/96010216-dd190680-0e41-11eb-915f-4142e9e01c6e.png)

Reply:
It should work with the latest version of typescript-eslint: `4.4.1`. VSCode may still use the old version because I did not update its internal linter state correctly when installing the new typescript-eslint version. When manually running `npm run lint` the error should not be visible.

Reply:
Thanks, `npm i` solved it..